tools/replace_code_block.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import sys
+from traceback import print_tb
+from typing import AnyStr
+
+
+def check_for_code(filename: AnyStr, check: bool, print_not_fixed: bool):
+    """Check a file and replace ``.. code::`` to ``.. code-block::``
+
+    :param filename: The file to open
+    :param check: Whether or not the program is operating in ``check`` mode.
+        In check mode, the program will exit with a status 1 if files will be
+        replaced. In non-check mode, the program will replace files.
+    :param print_not_fixed: Print files that were not re-formatted, similar to
+        black.
+    """
+    status = 0
+    with open(filename, "rb") as fp:
+        old_content = fp.read()
+    new_content = old_content.replace(b".. code::", b".. code-block::")
+    if old_content == new_content:
+        if print_not_fixed:
+            print(
+                "File {filename} does not contain any ``code::`` "
+                "statements".format(filename=filename)
+            )
+        return 0
+    if check:
+        status = 1
+        print(
+            "File {filename} contains ``code::`` statements!".format(
+                filename=filename
+            )
+        )
+    else:
+        with open(filename, "wb") as fp:
+            fp.write(new_content)
+        print(
+            "Replaced ``code::`` statements to ``code-block::`` in "
+            "{filename}".format(filename=filename)
+        )
+    return status
+
+
+def _main(check, verbose):
+    status = 0
+    for current_directory, directories, filenames in os.walk(
+        os.path.abspath(os.path.join(__file__, "..", "..", "praw"))
+    ):
+        for filename in filenames:
+            filename = os.path.join(current_directory, filename)
+            if filename.endswith(".py"):
```

Comment:
Can you do the `not ... continue` test instead of nesting logic within deeper blocks?

Reply:
I do not understand what a `not ... continue` test is

Reply:
https://github.com/praw-dev/praw/pull/1258#discussion_r363131588